### PR TITLE
[request for feedback] add two possible options of modeling case property choices

### PR DESCRIPTION
**Pull request by czue**
Not a real PR, but wanted to solicit feedback on the best way to model this. For [this spec](https://docs.google.com/document/d/1cE66puf6B_QSAsf4jkLKMVhwKt3fp3MKUh8Wx7uCFh0/edit#).

### CODE REVIEW

**Comment by orangejenny** [comment]
I like option 2: ability to do better validation seems valuable and it doesn't seem significantly harder to code/maintain.

Are you thinking of "no joins necessary" as an advantage for performance, code clarity, or other reasons?

Agree the ManyToMany probably isn't worth it.

File: corehq/apps/data_dictionary/models.py [hunk 1]
```diff
@@ -83,3 +90,17 @@ def save(self, *args, **kwargs):
         from .util import get_data_dict_props_by_case_type
         get_data_dict_props_by_case_type.clear(self.case_type.domain)
         return super(CaseProperty, self).save(*args, **kwargs)
+
+
+# option 2: normalized list of "choices" mapping to properties.
+# advantages: clearly structured in DB. easy to add more information to properties.
+# disadvantages: lots of joins.
+# note: could also make it a manytomany if we wanted to reuse property choices across properties
+# though not convinced that's worth it.
```
Comment (millerdev) [inline corehq/apps/data_dictionary/models.py:99]: I agree, many-to-many seems like overkill here.

**Review by millerdev — COMMENTED** [review]
+1 for normalized data. Joins can be optimized. JSONField seems like premature optimization unless we know for sure we will never want to query on that content.

**Review by proteusvacuum — COMMENTED** [review]
I also like option two, just from a clarity and standardization standpoint. 

I was under the impression that you can query a JSONField in postgres? Maybe we haven't made it to whatever version is required for that yet.

**Comment by snopoke** [comment]
Also agree with option 2 (without the many to many). 

Just a side note, if we used an actual JSON field in the DB we could index and query into it. I still don't think it's the right fit here but just pointing that out. From v3 Django supports some level of json querying: https://docs.djangoproject.com/en/3.1/topics/db/queries/#querying-jsonfield

**Comment by czue** [comment]
Thanks for the feedback everyone! Will go with Option 2.

**Pull request closed without merging**